src/opentime/timeRange.h

```diff
@@ -5,99 +5,246 @@
 #include <algorithm>
 #include <string>
 
-namespace opentime { namespace OPENTIME_VERSION {
-    
-class TimeRange {
-public:
-    explicit TimeRange() : _start_time {}, _duration {} {}
+namespace opentime {
+    namespace OPENTIME_VERSION {
+
+/**It is possible to construct TimeRange object with a negative duration.
+ * However, the logical predicates are written as if duration is positive,
+ * and have undefined behavior for negative durations.
+ *
+ * The duration on a TimeRange indicates a time range that is inclusive of the start time,
+ * and exclusive of the end time. All of the predicates are computed accordingly.
+ */
+
+        class TimeRange {
```

Comment:
minor point, but let's not change the indentation of the file in this PR. If we are changing the indentation, and the code, it is really hard to tell what is new and what is old.

Reply:
Will do. Thanks for the review!
Do we use clang_format or something? These indentation changes were done by the autoformat thing in CLion.

Reply:
AFAIK we haven't got a clang-format file, but that would be a good thing to have ultimately. Perhaps CLion is using a standard format if none is found in the project?

Reply:
added an issue for having a clang format file